tools/icqsol_translate_shape/icqsol_translate_shape.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+import argparse
+import shutil
+
+from icqsol.shapes.icqShapeManager import ShapeManager
+from icqsol import util
+import icqsol_utils
+
+# Parse Command Line.
+parser = argparse.ArgumentParser()
+parser.add_argument('--input', dest='input', help='Shape dataset selected from history')
+parser.add_argument('--input_file_format_and_type', dest='input_file_format_and_type', help='Input file format and type')
+parser.add_argument('--input_dataset_type', dest='input_dataset_type', help='Input dataset_type')
+parser.add_argument('--displacement_x', dest='displacement_x', type=float, default=1.0, help='X coordinate of displacement')
+parser.add_argument('--displacement_y', dest='displacement_y', type=float, default=0.0, help='Y coordinate of displacement')
+parser.add_argument('--displacement_z', dest='displacement_z', type=float, default=0.0, help='Z coordinate of displacement')
+parser.add_argument('--output', dest='output', help='Output file name')
+parser.add_argument('--output_vtk_type', dest='output_vtk_type', default='ascii', help='Output VTK type')
+
+args = parser.parse_args()
+
+# Get the format of the input - either vtk or ply.
+input_format, input_file_type = icqsol_utils.get_format_and_type(args.input_file_format_and_type)
+tmp_dir = icqsol_utils.get_temp_dir()
+
+# Get the format of the input - either vtk or ply.
+file_format = util.getFileFormat(args.input)
+
+# Instantiate a ShapeManager for loading the input.
+if file_format == util.VTK_FORMAT:
+    shape_mgr = ShapeManager(file_format=file_format, vtk_dataset_type=args.input_dataset_type)
+else:
+    shape_mgr = ShapeManager(file_format=file_format)
+
+# Get the vtk polydata from the input dataset.
+vtk_poly_data = shape_mgr.loadAsVtkPolyData(args.input)
+
+displ = (args.displacement_x, args.displacement_y, args.displacement_z)
+
+# Translate (in place operation).
```

Comment:
@gregvonkuster

You're looking at an old file...

# Get the format of the input - either vtk or ply.input_format,
input_file_type =
icqsol_utils.get_format_and_type(args.input_file_format_and_type)tmp_dir =
icqsol_utils.get_temp_dir() # Instantiate a ShapeManager for loading the
input.if input_format == 'vtk': shape_mgr =
ShapeManager(file_format=input_format,
vtk_dataset_type=args.input_dataset_type)else: shape_mgr = ShapeManager(
file_format=input_format)

On Thu, Dec 3, 2015 at 11:11 AM, Greg Von Kuster <notifications@github.com>
wrote:

> In tools/icqsol_translate_shape/icqsol_translate_shape.py
> <https://github.com/gregvonkuster/galaxy-csg/pull/18#discussion_r46588206>
> :
>
> > +
> > +# Get the format of the input - either vtk or ply.
> > +file_format = util.getFileFormat(args.input)
> > +
> > +# Instantiate a ShapeManager for loading the input.
> > +if file_format == util.VTK_FORMAT:
> > +    shape_mgr = ShapeManager(file_format=file_format, vtk_dataset_type=args.input_dataset_type)
> > +else:
> > +    shape_mgr = ShapeManager(file_format=file_format)
> > +
> > +# Get the vtk polydata from the input dataset.
> > +vtk_poly_data = shape_mgr.loadAsVtkPolyData(args.input)
> > +
> > +displ = (args.displacement_x, args.displacement_y, args.displacement_z)
> > +
> > +# Translate (in place operation).
>
> @pletzer <https://github.com/pletzer> Getting close. After the above
> issue is handled, you need to add a .shed.yml file for each tool. These
> files should be modeled after the existing .shed.yml files associated with
> the existing tools. You should validate the contents of your .shed.yml file
> with a YAML validator - you can find one via Google (e.g.,
> http://yaml-online-parser.appspot.com).
>
> These .shed.yml files are used by Planemo (https://planemo.readthedocs.org)
> to help administer the tool and its associated files over time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gregvonkuster/galaxy-csg/pull/18/files#r46588206>.
>



-- 
Alexander Pletzer
+1 609 799 7999 cell
alexander@gokliya.net